src/Spd.Resource.Repository.IntegrationTest/ControllingMemberCrcRepositoryTest.cs

```diff
@@ -0,0 +1,396 @@
+            .With(a => a.ControllingMemberAppId, controllingMemberAppId)
+            .Create();
+        contact? contact = new contact()
+        {
+            contactid = cmd.ContactId,
+        };
+        _context.AddTocontacts(contact);
+        // Create Account, then set biz app's account
+        account account = new account()
+        {
+            accountid = accountid,
+        };
+        _context.AddToaccounts(account);
+        // Create Biz Licence app
+        spd_application bizApp = new spd_application()
+        {
+            spd_applicationid = bizlicAppId
+        };
+        _context.AddTospd_applications(bizApp);
+        _context.SetLink(bizApp, nameof(spd_application.spd_ApplicantId_account), account);
+        await _context.SaveChangesAsync();
+        // Create Biz Contact
+        spd_businesscontact bizContact = new spd_businesscontact()
+        {
+            spd_businesscontactid = bizContactId,
+            statecode = DynamicsConstants.StateCode_Active
+        };
+        _context.AddTospd_businesscontacts(bizContact);
+        _context.AddLink(bizContact, nameof(bizContact.spd_businesscontact_spd_application), bizApp);
+        await _context.SaveChangesAsync();
+
+        // Action
+        try
+        {
+            ControllingMemberCrcApplicationCmdResp? resp = await _controllingMemberCrcRepository.SaveControllingMemberCrcApplicationAsync(cmd, CancellationToken.None);
+            controllingMemberAppId = resp.ControllingMemberAppId;
+            // Assert
+            ControllingMemberCrcApplicationResp getResult = await _controllingMemberCrcRepository.GetCrcApplicationAsync(resp.ControllingMemberAppId, CancellationToken.None);
+            Assert.NotNull(getResult);
+            Assert.Equal(getResult.GivenName, cmd.GivenName);
+            Assert.Equal(getResult.Surname, cmd.Surname);
+            Assert.Equal(getResult.DateOfBirth, cmd.DateOfBirth);
+            Assert.Equal(getResult.MiddleName1, cmd.MiddleName1);
+            Assert.Equal(getResult.GenderCode, cmd.GenderCode);
+            Assert.Equal(getResult.PhoneNumber, cmd.PhoneNumber);
+            Assert.Equal(getResult.Aliases, cmd.Aliases);
+            Assert.NotNull(getResult.CaseNumber);
+        }
+        catch (Exception e)
+        {
+            throw e.GetBaseException();
+        }
+        finally
+        {
+            //Clean up objects
+            await CleanUpObject(contactId, accountid, bizlicAppId, bizContactId, controllingMemberAppId);
+        }
+    }
+
+    private async Task CleanUpObject(Guid? contactId, Guid? accountid, Guid? bizlicAppId, Guid? bizContactId, Guid? controllingMemberAppId)
+    {
+        spd_application? appToRemove = _context.spd_applications.Where(a => a.spd_applicationid == controllingMemberAppId).SingleOrDefault();
+
+        if (appToRemove != null)
+        {
+            _context.DeleteObject(appToRemove);
+            await _context.SaveChangesAsync();
```

Comment:
do not save changes everytime. 
Remove it. at the end of function, add saveChanges.

Reply:
this way it causes exception

Reply:
what exception?